src/overlay006/ov6_02240C9C.c

```diff
@@ -1406,142 +1285,122 @@ void ov6_02242034 (FieldSystem * fieldSystem, BattleParams * param1)
+        return FALSE;
     }
 }
 
-static BOOL ov6_0224219C (const UnkStruct_ov6_0224222C * param0, const u8 param1, const u8 param2, u8 * param3)
+static BOOL ov6_0224219C (const UnkStruct_ov6_0224222C * param0, const u8 maxEncounters, const u8 param2, u8 * encounterSlot)
 {
-    u8 v0[12];
-    u8 v1;
+    u8 v0[MAX_GRASS_ENCOUNTERS];
     u8 v2;
-    u8 v3, v4;
 
-    for (v2 = 0; v2 < 12; v2++) {
+    for (v2 = 0; v2 < MAX_GRASS_ENCOUNTERS; v2++) {
         v0[v2] = 0;
     }
 
-    v1 = 0;
+    u8 v1 = 0;
 
-    for (v2 = 0; v2 < param1; v2++) {
-        v3 = PokemonPersonalData_GetSpeciesValue(param0[v2].unk_00, 6);
-        v4 = PokemonPersonalData_GetSpeciesValue(param0[v2].unk_00, 7);
+    for (v2 = 0; v2 < maxEncounters; v2++) {
+        u8 v3 = PokemonPersonalData_GetSpeciesValue(param0[v2].species, MON_DATA_PERSONAL_TYPE_1);
+        u8 v4 = PokemonPersonalData_GetSpeciesValue(param0[v2].species, MON_DATA_PERSONAL_TYPE_2);
 
         if ((v3 == param2) || (v4 == param2)) {
             v0[v1++] = v2;
         }
     }
 
-    if ((v1 == 0) || (v1 == param1)) {
-        return 0;
+    if ((v1 == 0) || (v1 == maxEncounters)) {
+        return FALSE;
     }
 
-    *param3 = v0[LCRNG_Next() % v1];
-    return 1;
+    *encounterSlot = v0[LCRNG_Next() % v1];
+    return TRUE;
 }
 
-static BOOL ov6_0224222C (Pokemon * param0, const UnkStruct_ov6_022422D0 * param1, const UnkStruct_ov6_0224222C * param2, const u8 param3, const u8 param4, const u8 param5, u8 * param6)
+static BOOL ov6_0224222C (Pokemon * param0, const WildEncounters_FieldParams * param1, const UnkStruct_ov6_0224222C * param2, const u8 maxEncounters, const u8 param4, const u8 ability, u8 * encounterSlot)
 {
-    if (param1->unk_0D == 0) {
-        if (param1->unk_0E == param5) {
-            if (inline_020564D0(2) == 0) {
-                return ov6_0224219C(param2, param3, param4, param6);
-            }
-        }
+    if ((param1->isFirstMonEgg == FALSE) && (param1->firstMonAbility == ability) && (inline_020564D0(2) == 0)) {
+                return ov6_0224219C(param2, maxEncounters, param4, encounterSlot);
     }
 
-    return 0;
+    return FALSE;
 }
 
-static u8 ov6_0224226C (const BOOL param0, const u8 param1, const UnkStruct_ov6_022422D0 * param2, const u32 param3, Pokemon * param4)
+static u8 ov6_0224226C (const BOOL isFishingEncounter, const u8 param1, const WildEncounters_FieldParams * param2, const u32 weatherEffect, Pokemon * param4)
 {
-    int v0;
-
-    v0 = param1;
+    int v0 = param1;
 
-    if (param2->unk_0D == 0) {
-        if (param0) {
-            if ((param2->unk_0E == 60) || (param2->unk_0E == 21)) {
-                (void)(v0 * 2);
+    if (!param2->isFirstMonEgg) {
+        if (isFishingEncounter) {
+            if ((param2->firstMonAbility == ABILITY_STICKY_HOLD) || (param2->firstMonAbility == ABILITY_SUCTION_CUPS)) {
+                v0 * 2; // BUG: Abilities do not Increase Fishing Encounter Rate (see docs/bugs_and_glitches.md)
```

Comment:
**question:** Does this have an entry in `docs/bugs_and_glitches.md`?

Reply:
Added with 8ceccf4